### PR TITLE
test/e2e: NamespacedTest helper can create/delete multiple namespaces

### PR DESCRIPTION
**Pull request by satyazzz123**
changed the function signature from func `(f *Framework) NamespacedTest(namespace string, body NamespacedTestBody)` to this
`func (f *Framework) NamespacedTest(namespace string, body NamespacedTestBody, additionalNamespaces ...string)`

Updates: #3837

### CODE REVIEW

File: test/e2e/framework.go [hunk 1]
```diff
@@ -283,7 +283,7 @@ type NamespacedGatewayTestBody func(ns string, gw types.NamespacedName)
 type NamespacedTestBody func(string)
 type TestBody func()
 
-func (f *Framework) NamespacedTest(namespace string, body NamespacedTestBody) {
+func (f *Framework) NamespacedTest(namespace string, body NamespacedTestBody, additionalNamespaces ...string) {
 	ginkgo.Context("with namespace: "+namespace, func() {
 		ginkgo.BeforeEach(func() {
 			f.CreateNamespace(namespace)
```
Comment (sunjayBhatia) [inline test/e2e/framework.go:289]: we also want to make sure these additional namespaces are created and deleted in the before/aftereach blocks

Comment (satyazzz123) [reply test/e2e/framework.go]: > we also want to make sure these additional namespaces are created and deleted in the before/aftereach blocks

I have made some changes , please have a look

**Comment by satyazzz123** [comment]
@sunjayBhatia is this good to merge? Or I need to make some changes??

**Review by sunjayBhatia — COMMENTED** [review]
It would be great to also *use* this helper where it is needed in the e2e tests to fully finish https://github.com/projectcontour/contour/issues/3837

e.g. you can search for `CreateNamespace` in the `test/e2e` directory to find where we're manually creating extra namespaces that could be transferred to using this helper

If you want to do that in this PR that would be great, or can be done in a follow up

**Comment by satyazzz123** [comment]
> It would be great to also _use_ this helper where it is needed in the e2e tests to fully finish #3837
> 
> e.g. you can search for `CreateNamespace` in the `test/e2e` directory to find where we're manually creating extra namespaces that could be transferred to using this helper
> 
> If you want to do that in this PR that would be great, or can be done in a follow up

If it's not any issue then I can open another PR for the adding the e2e tests using this helper and we can merge this one for the namespaced test. I am working on it right now

**Review by sunjayBhatia — APPROVED** [review]

**Pull request merged**